### PR TITLE
Adding helm deployment/pipeline for Maintenance Page

### PR DESCRIPTION
**Pull request by brero12**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?
On local env

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by pbolduc** [comment]
@brero12  could you move the new chart out from the nested under `gitops/charts/traffic-court-online`. I would like the maintenance-site chart as a separate deployable / updatable chart. 

Also, could you add parameter for the route to allow it to optionally created. I like being able to test charts on non-OpenShift kubernetes like Docker Desktop Kubernetes. 

Generally this looking good.

File: gitops/charts/traffic-court-online/charts/maintenance-site/templates/route.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+kind: Route
```
Comment (pbolduc) [inline gitops/charts/traffic-court-online/charts/maintenance-site/templates/route.yaml:1]: add a `{{- if .Values.route.create }}` and a value that defaults to `true`

File: gitops/charts/traffic-court-online/charts/maintenance-site/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      annotations:
+        checksum/keycloakConfig: {{ include "maintenance-site.tplvalues.render" ( dict "value" .Values.keycloakConfig "context" $ ) | sha256sum }}
+      labels:
+        {{- include "maintenance-site.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ .Values.global.licenseplate }}-vault
```
Comment (pbolduc) [inline gitops/charts/traffic-court-online/charts/maintenance-site/templates/deployment.yaml:28]: I dont think we need to use the vault service account. This app will not use secrets if I understand correctly.

File: gitops/charts/traffic-court-online/charts/maintenance-site/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "1.16.0"
```
Comment (pbolduc) [inline gitops/charts/traffic-court-online/charts/maintenance-site/Chart.yaml:24]: App version could probably be 1.0.0

File: gitops/charts/traffic-court-online/charts/maintenance-site/templates/route.yaml (hunk 1 above)
Comment (brero12) [reply gitops/charts/traffic-court-online/charts/maintenance-site/templates/route.yaml]: Made it according to suggestions.

File: gitops/charts/traffic-court-online/charts/maintenance-site/templates/deployment.yaml (hunk 2 above)
Comment (brero12) [reply gitops/charts/traffic-court-online/charts/maintenance-site/templates/deployment.yaml]: Removed

File: gitops/charts/traffic-court-online/charts/maintenance-site/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "maintenance-site.selectorLabels" . | nindent 6 }}
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations:
+        checksum/keycloakConfig: {{ include "maintenance-site.tplvalues.render" ( dict "value" .Values.keycloakConfig "context" $ ) | sha256sum }}
```
[the remark this answers is not in the document]
Comment (brero12) [reply gitops/charts/traffic-court-online/charts/maintenance-site/templates/deployment.yaml]: this is a mistake when I followed template . Removed.

File: gitops/charts/traffic-court-online/charts/maintenance-site/Chart.yaml (hunk 3 above)
Comment (brero12) [reply gitops/charts/traffic-court-online/charts/maintenance-site/Chart.yaml]: Made it according to suggestions.

**Review by pbolduc — APPROVED** [review]
LGTM - Thank you

**Pull request merged**